python_modules/dagster/dagster/_core/execution/context/data_version_cache.py

```diff
@@ -93,54 +93,83 @@ def fetch_external_input_asset_version_info(self) -> None:
                     all_dep_keys.append(key)
 
         self.input_asset_version_info = {}
-        for key in all_dep_keys:
-            self._fetch_input_asset_version_info(key)
+        self._fetch_input_asset_version_info(all_dep_keys)
```

Comment:
Do we need to limit the batch size here in some way and still potentially do more than one call? I think in the specific example that prompted this change the total number of input keys was impressively large

Reply:
yeah, that's worth calling out.... this is still wip